FHEM/00_SIGNALduino.pm

```diff
@@ -2216,221 +2237,224 @@ sub SIGNALduino_Parse_MU($$$$@)
 	
 #			$valid=SIGNALduino_inTol($ProtocolListSIGNALduino{$id}{clockabs},$clockabs,$clockabs*0.30) if ($ProtocolListSIGNALduino{$id}{clockabs} > 0);
 
-			next if (!$valid) ;
 			
-			my $bit_length = ($signal_length/((scalar @{$ProtocolListSIGNALduino{$id}{one}} + scalar @{$ProtocolListSIGNALduino{$id}{zero}})/2));
-			Debug "Expect $bit_length bits in message"  if ($valid && $debug);
-
-			#Check calculated min length
-			#$valid = $valid && $ProtocolListSIGNALduino{$id}{length_min} <= $bit_length if (exists $ProtocolListSIGNALduino{$id}{length_min}); 
-			#Check calculated max length
-			#$valid = $valid && $ProtocolListSIGNALduino{$id}{length_max} >= $bit_length if (exists $ProtocolListSIGNALduino{$id}{length_max});
-
-			#next if (!$valid) ;
-			#Debug "expecting $bit_length bits in signal" if ($debug && $valid);
+			#my $bit_length = ($signal_length/((scalar @{$ProtocolListSIGNALduino{$id}{one}} + scalar @{$ProtocolListSIGNALduino{$id}{zero}})/2));
+			#Debug "Expect $bit_length bits in message"  if ($debug);
 
 			Debug "Searching in patternList: ".Dumper(\%patternList) if($debug);
-			next if (!$valid) ;
 
-	
+			my @msgStartLst;
+			my $startStr=""; # Default match if there is no start pattern available
+			my $message_start=0 ;
+			my $startLogStr="";
+			
+			if (defined($ProtocolListSIGNALduino{$id}{start}))	# wenn start definiert ist, dann startStr ermitteln und in rawData suchen und in der rawData alles bis zum startStr abschneiden
+			{
+				@msgStartLst = $ProtocolListSIGNALduino{$id}{start};
+				Debug "msgStartLst: ".Dumper(@msgStartLst)  if ($debug);
+				
+				if ( ($startStr=SIGNALduino_PatternExists($hash,@msgStartLst,\%patternList,\$rawData)) eq -1)
+				{
+					Log3 $name, 5, "$name: start pattern for MU Protocol id $id -> $ProtocolListSIGNALduino{$id}{name} not found, aborting" if ($dummy);
+					next;
+				}
+				Debug "startStr is: $startStr" if ($debug);
+				
+				$message_start = index($rawData, $startStr);
+				if ($message_start >= 0) {
+					$rawData = substr($rawData, $message_start);
```

Comment:
@sidey79 da hast Du anscheinend in Deiner Version zu viel optimiert, bei meiner ursprünglicher Version hat es noch gepasst
https://github.com/RFD-FHEM/RFFHEM/issues/381